skll/utilities/generate_predictions.py

```diff
@@ -187,21 +92,47 @@ def main(argv=None):
                                 '%(message)s'))
     logger = logging.getLogger(__name__)
 
-    if args.positive_label:
-        logger.warning("The `positive_label` argument is deprecated. Use "
-                       "`positive_label_index` instead.")
-        positive_label_index = args.positive_label
-    else:
-        positive_label_index = args.positive_label_index
-
-    # Create the classifier and load the model
-    predictor = Predictor(args.model_file,
-                          positive_label_index=positive_label_index,
-                          threshold=args.threshold,
-                          all_labels=args.all_probabilities,
-                          logger=logger)
-
-    # Iterate over all the specified input files
+    # load the model from disk
+    learner = Learner.from_file(args.model_file)
+
+    # is the model a regressor or a classifier?
+    estimator_type = learner.model_type._estimator_type
+
+    # if we are using a binary classification model, get the positive
+    # class label string if from the `pos_label_str` attribute, and if
+    # that is `None`, get it from the learner's label dictionary; also
+    # get the string denoting the negative label which is just the other
+    # label in the list
+    if estimator_type == 'classifier':
+        if len(learner.label_list) == 2:
+            if learner.pos_label_str:
```

Comment:
Could `pos_label_str` still be 0 here? If so, shouldn't this also be `if learner.pos_label_str is not None`?

Reply:
Good catch!